app/src/main/java/org/keepgoeat/presentation/home/HomeActivity.kt

```diff
@@ -110,8 +125,21 @@ class HomeActivity : BindingActivity<ActivityHomeBinding>(R.layout.activity_home
         finish()
     }
 
+    private fun compareVersion(currentVersion: String, updateVersion: String): Boolean {
+        val splitCurrent = currentVersion.split(".")
+        val splitUpdate = updateVersion.split(".")
+
+        for (i in 0..2) {
+            if (splitCurrent[i].toInt() < splitUpdate[i].toInt())
+                return true
+        }
+        return false
+    }
```

Comment:
```suggestion
    private fun compareVersion(updateVersion: String): Boolean {
        val splitCurrent = BuildConfig.VERSION_NAME.split(".")
        val splitUpdate = updateVersion.split(".")

        for (i in 0..2) { // 범위 수정 필요!
            if (splitCurrent[i].toInt() < splitUpdate[i].toInt())
                return true
        }
        return false
    }
```

1. 현재 버전 정보는 **파라미터를 전달받지 않아도 괜찮을 것 같아요**! BuildConfig.VERSION_NAME으로 어디서나 접근 가능하기 때문입니다~ 

2. 그리고 혹시나 1.0과 같이 두자리로 변경(현재는 1.0.0)될 가능성도 있기 때문에 **반복문의 범위를 0..2로 지정하게되면 의도치 않은 버그가 발생**할 수 있습니다. 따라서 해당 로직을 다음과 같은 로직으로 수정해볼 수도 있겠어요!
두개의 버전 정보를 비교해서 사이즈가 더 작은 버전 정보를 기준으로 반복할 인덱스의 범위를 정하되, 마지막 인덱스에 해당하는 숫자까지 동일해서 대소비교가 불가하다면 사이즈가 더 큰 버전정보 리스트를 반환합니다. 사실 여기서도 고려해야할 부분은 1.0과 1.0.0인 경우인데요! 1.0.0이 사이즈가 3이지만 마지막 숫자가 0이기 때문에 동일한 버전으로 간주하는 처리가 필요할 것 같습니다...! 하하하.. 해당 로직은 참고용이니까 그대로 따르지 않으셔도 괜찮아용!

3. 데이터 관련 로직에 가까우니까 **뷰모델로** 옮기고, 액티비티에서 뷰모델의 함수를 호출해보는 건 어떨가용?!